app.rb

```diff
@@ -154,12 +159,13 @@ def create_rental_with_book_and_person(input_book, input_person)
     print 'Date in format YYYY-MM-DD:  '
     date = @user_interface.user_input
     if /^\d{4}-\d{2}-\d{2}$/.match(date)
+      rental = Rental.new(date)
       selected_book = @books[input_book]
       selected_person = @persons[input_person]
-      rental = Rental.new(date)
       rental.add_book(selected_book)
       rental.add_person(selected_person)
       @rentals << rental
```

Comment:
- [ ] Great job so far team. You have done an amazing job so far and your app is able to preserve data as expected :clap: 
- [ ] However, I deleted all the current `JSON` files in your project to test if the part of the project requirement that mentioned that you need to remember to handle the case when one or more of the files are missing and ensure that the app works and shows no errors 👍 
- [ ] The app did start as expected. and when some inputs were made to create a new `person` and `book`, the respective `JSON files `were recreated and stored the data without any errors :clap: However, in an attempt to create a new `rental`, the app threw an error and stopped working. :worried: 
- [ ] As a result, data wasn't completed persisted in your app as expected so kindly fix this and ensure that your app works as expected even if any of the `JSON files` are missing 😉 
- Don't forget to also ensure that it is possible to `List all rentals for a given person's ID` :+1: 
- [ ] If you look at the video, you'll notice that the `Create a rental` and `List all rentals for a given person's ID` features didn't work as expected :+1:  So kindly work on this so your app works as expected :100: 
- Video Reference
_**I took this video after deleting all the JSON files from the project 👍**_

[Screencast from 07-12-2023 04:30:02 PM.webm](https://github.com/muneeb3123/library-books/assets/102699958/085dfd2c-682b-4585-900a-c1778eb421d3)

- Project Requirement
![image](https://github.com/dev89s/school-library/assets/102699958/dc3da349-ee7c-4ca1-853d-7f0f8f610931)

Reply:
done now it works